From 518c10bb3ec0f6cb3efed01e43902851af28224e Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sre@kernel.org>
Date: Sun, 5 Oct 2025 01:21:22 +0200
Subject: [PATCH 102/263] test T14s EC suspend bits

---
 drivers/platform/arm64/lenovo-thinkpad-t14s.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/platform/arm64/lenovo-thinkpad-t14s.c b/drivers/platform/arm64/lenovo-thinkpad-t14s.c
index 5590302a5..e8a29ed8f 100644
--- a/drivers/platform/arm64/lenovo-thinkpad-t14s.c
+++ b/drivers/platform/arm64/lenovo-thinkpad-t14s.c
@@ -3,6 +3,8 @@
  * Copyright (c) 2025, Sebastian Reichel
  */
 
+#define DEBUG
+
 #include <linux/bitfield.h>
 #include <linux/bits.h>
 #include <linux/cleanup.h>
@@ -484,6 +486,10 @@ static irqreturn_t t14s_ec_irq_handler(int irq, void *data)
 		return IRQ_HANDLED;
 	}
 
+	pm_wakeup_event(ec->dev, 0);
+
+	// TODO: pm_system_cancel_wakeup() for T14S_EC_EVT_LID_CLOSED and T14S_EC_EVT_AC_DISCONNECTED?
+
 	switch (val) {
 	case T14S_EC_EVT_NONE:
 		break;
@@ -599,7 +605,8 @@ static int t14s_ec_probe(struct i2c_client *client)
 	/*
 	 * Disable wakeup support by default, because the driver currently does
 	 * not support masking any events and the laptop should not wake up when
-	 * the LID is closed.
+	 * the LID is closed. Unfortunately writing to the T14S_EC_REG_MODERN_STANDBY
+	 * register is not enough either.
 	 */
 	device_wakeup_disable(dev);
 
-- 
2.43.0

